packages/ra-ui-materialui/src/field/FileField.tsx

```diff
@@ -114,16 +118,13 @@ export interface FileFieldProps<
     RecordType extends Record<string, any> = Record<string, any>,
 > extends FieldProps<RecordType> {
     src?: string;
-    title?: Call<Objects.AllPaths, RecordType> extends never
-        ? AnyString
-        : Call<Objects.AllPaths, RecordType> | AnyString;
+    title?: HintedString<ExtractRecordPaths<RecordType>>;
```

Comment:
This is a regression: we used to accept AnyString. Same for Imagefield.

Reply:
No `HintedString` does the same